allennlp/models/semantic_parsing/nlvr/nlvr_decoder_step.py

```diff
@@ -132,10 +127,12 @@ def take_step(self,  # type: ignore
         # action_mask: (group_size, num_embedded_actions)
         action_embeddings, embedded_action_mask = self._get_action_embeddings(state,
                                                                               global_actions_to_embed)
-        action_query = self._get_action_query(state, hidden_state, attended_sentence)
+        action_query = torch.cat([hidden_state, attended_sentence], dim=-1)
         # (group_size, action_embedding_dim)
         predicted_action_embedding = self._output_projection_layer(action_query)
-
+        if state.checklist_state[0] is not None:
+            embedding_addition = self._get_predicted_embedding_addition(state)
+            predicted_action_embedding += self._checklist_embedding_multiplier * embedding_addition
```

Comment:
Don't ever use `+=` on a tensor.  It doesn't do what you expect.  Use `x = x + y` instead.

Reply:
Yes, I figured out that `+=` causes in-place updates, messing up the computation graph while `x = x+y` does not. I changed the other places I did in-place updates, but I guess I missed these two. Thanks, fixed them.